tests/waku_rln_relay/test_rln_group_manager_onchain.nim

```diff
@@ -37,56 +37,91 @@ proc generateCredentials(rlnInstance: ptr RLN, n: int): seq[IdentityCredential]
     credentials.add(generateCredentials(rlnInstance))
   return credentials
 
-#  a util function used for testing purposes
-#  it deploys membership contract on Ganache (or any Eth client available on EthClient address)
-#  must be edited if used for a different contract than membership contract
-proc uploadRLNContract*(ethClientAddress: string): Future[Address] {.async.} =
-  let web3 = await newWeb3(ethClientAddress)
-  debug "web3 connected to", ethClientAddress
-
-  # fetch the list of registered accounts
-  let accounts = await web3.provider.eth_accounts()
-  web3.defaultAccount = accounts[1]
-  let add = web3.defaultAccount
-  debug "contract deployer account address ", add
+when defined(rln_v2):
+  #  a util function used for testing purposes
+  #  it deploys membership contract on Ganache (or any Eth client available on EthClient address)
+  #  must be edited if used for a different contract than membership contract
+  # <the difference between this and rln-v1 is that there is no need to deploy the poseidon hasher contract>
+  proc uploadRLNContract*(ethClientAddress: string): Future[Address] {.async.} =
+    let web3 = await newWeb3(ethClientAddress)
+    debug "web3 connected to", ethClientAddress
+
+    # fetch the list of registered accounts
+    let accounts = await web3.provider.eth_accounts()
+    web3.defaultAccount = accounts[1]
+    let add = web3.defaultAccount
+    debug "contract deployer account address ", add
+
+    let balance = await web3.provider.eth_getBalance(web3.defaultAccount, "latest")
+    debug "Initial account balance: ", balance
+
+    # deploy registry contract with its constructor inputs
+    let receipt = await web3.deployContract(RegistryContractCode)
```

Comment:
This is the only difference between the two branches of the `when defined(rln_v2)`. Thereafter, I think is better to perform the `when` validation in a narrower scope instead of duplicating so much code.

Reply:
will make that change 👍 
wanted to make it easier to transition from v1 to v2 by deleting whole functions

Reply:
addressed in b05815e3